### DISCUSSION TITLE
How to modify a compose key sequence?

### DISCUSSION THREAD

**Discussion by charmoniumQ** [hyprwm/Hyprland#4639] [Q&A] [answered]
I have `kb_options=compose:ralt`, so I can type `right_alt + < + >` and hyprland inserts a `⋄` (diamond operator).

I would rather have this same key combination insert a `◊` (lozenge), but keep the other compose key sequences?

In X11, you could modify `~/.XCompose`. What's the hyprland equivalent?

**Comment by MHellmund** [comment, marked as answer]
> In X11, you could modify ~/.XCompose. What's the hyprland equivalent?

The same. Wayland uses ~/.XCompose, when present.

**Reply by tizu69** [reply]
how would I hot reload this?

**Reply by bemmesr** [reply]
> > In X11, you could modify ~/.XCompose. What's the hyprland equivalent?
> 
> The same. Wayland uses ~/.XCompose, when present.

Not quite the same, actually. Though most applications do seem to respect the custom compose sequences in .XCompose under Hyprland, there are some that don't. This is not the same behaviour as in X11, wherein *all* applications are force‐fed the composed keys instead of the raw input.

**Comment by Mikilio** [comment]
On NixOS this doesn't quite seem to be working. Can anyone confirm?

**Reply by WhereAreWeSeeing** [reply]
I have this exact problem on nixos, did you ever get it to work?

**Reply by charmoniumQ** [reply]
Mine works, I use home-manager to deploy `~/.XCompose` on NixOS 26.11

**Reply by Mikilio** [reply]
My problem all these years was that my XCompse had a syntax error.
